internal/sink/kafka_connect.go

```diff
@@ -0,0 +1,57 @@
+package sink
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type KafkaConnect struct {
+	config *KafkaConnectConfig
+}
+
+type KafkaConnectConfig struct {
+	Address string
+}
+
+type Connector struct {
+	Name   string            `json:"name"`
+	Config map[string]string `json:"config"`
+}
+
+func NewKafkaConnect(config *KafkaConnectConfig) (*KafkaConnect, error) {
+	return &KafkaConnect{
+		config: config,
+	}, nil
+}
+
+func (k *KafkaConnect) CreateConnector(ctx context.Context, connector *Connector) error {
+	endpoint := fmt.Sprintf("%s/connectors", k.config.Address)
+
+	jsonData, err := json.Marshal(connector)
+	if err != nil {
+		return err
+	}
+
+	request, err := http.NewRequest("POST", endpoint, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return err
+	}
+	request = request.WithContext(ctx)
+	request.Header.Set("Content-Type", "application/json; charset=UTF-8")
+
+	client := &http.Client{}
```

Comment:
If anything, just use the http.DefaultClient, but better yet accept an http client in the constructor and fall back to the default if none provided.

The default client does not specify any timeouts whatsoever, so in the future we certainly want to set a custom HTTP client.